assets/js/modules/analytics/components/module/ModuleAcquisitionChannelsWidget/PieChart.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * PieChart component.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import GoogleChart from '../../../../../components/GoogleChart';
+import { extractAnalyticsDataForPieChart } from '../../../util';
+import PreviewBlock from '../../../../../components/PreviewBlock';
+
+export default function PieChart( { report, hasFinishedResolution } ) {
+	const processedData = extractAnalyticsDataForPieChart( report, { keyColumnIndex: 1 } );
+
+	return hasFinishedResolution ? (
+		<div className="googlesitekit-chart googlesitekit-chart--pie">
+			<GoogleChart
+				data={ processedData }
+				options={ PieChart.options }
+				chartType="pie"
+				id="overview-piechart"
+				loadHeight={ 205 }
+			/>
+		</div>
+	) : (
+		<PreviewBlock width="282px" height="282px" shape="circular" />
```

Comment:
Let's better keep preview related components in one place instead of spreading it across a couple of components even though it will require to repeat layout. In this case, it will be easier to maintain it in the future and if need be, we can update the preview state by changing it in one location.

```javascript
export default function ModuleAcquisitionChannelsWidget( ... ) {
    ...

    if ( ! hasFinishedResolution ) {
        return (
            <Widget Header={ Header } Footer={ Footer }>
                <Grid>
                    <Row>
                        <Cell size={ 4 }>
                            <PreviewBlock width="282px" height="282px" shape="circular" />
                        </Cell>
                        <Cell size={ 8 }>
                            <PreviewTable rows={ 4 } rowHeight={ 50 } />
                        </Cell>
                    </Row>
                </Grid>
            </Widget>
        );
    }

    if ( error ) {
        return ...;
    }

    if ( isZeroReport( report ) ) {
        return ...;
    }

    return ...;
}
```

Reply:
Updated in https://github.com/google/site-kit-wp/pull/3035/commits/ed7f018c7443539a70a7de9a6ca460f1d2f2c848